common/src/main/java/com/wynntils/core/text/StyledText.java

```diff
@@ -241,11 +242,17 @@ public StyledText getNormalized() {
     }
 
     public StyledText trim() {
-        return new StyledText(
-                parts.stream().map(StyledTextPart::trim).collect(Collectors.toList()),
-                temporaryWorkaround,
-                clickEvents,
-                hoverEvents);
+        if (parts.isEmpty()) {
+            return new StyledText(List.of(), temporaryWorkaround, clickEvents, hoverEvents);
+        }
+
+        List<StyledTextPart> newParts = new ArrayList<>(parts);
+        newParts.set(0, newParts.get(0).trim());
```

Comment:
Unfortunately, it is not that simple :( This will also trim trailing spaces of the first segment. You need to have a "trimLeadingSpacedOnly" functionality here. Unless Apache COmmons has something for that, you most likely need to reimplement Java's trim function.

Reply:
Java 11 for the win!